### PR TITLE
Ruby project#2: My enumerable methods

### PR DESCRIPTION
**Pull request by vanheaven-ui**
Contained herein is ruby code for my version of some of the ruby enumerable methods that apply to collection objects.

<li>The my_each and my_each_with_index methods are created by using an until loop to create indices and using these indices to enumerate through all the elements of the collection.</li>

<li>The rest of the methods created are based on the my_each method to enumerate through the elements of the collection.</li>

<li>The yield keyword is used to pass the blocks to the respective methods.</li>
<p></P>

Your review of the code is of great value to my understanding of the referenced subject 🤝.

Thanks and Regards
Ezekiel Mworekwa

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hi @vanheaven-ui  👋

You've done quite well on this, to get the project to its present level at the first review.🚀

Let's fix these few issues and we would be good to go.

Most of the issues have to do with my_none method:

- [ ] my_none should be structured to receive a Ruby class and return a result. Right now my_none returns nil in this example. [Screenshot](https://user-images.githubusercontent.com/33323077/82895722-9129fb80-9f4c-11ea-8e2d-4af2f2c1885e.png
)

- Also in my_none when a Regex is passed to it, it should return a result but right now yours returns nil. It fails in this example. [Screenshot](https://user-images.githubusercontent.com/33323077/82896018-20cfaa00-9f4d-11ea-86a9-d7dad7266f26.png)

- In addition when a simple pattern is passed to my_none it still returns nil. Take an example like this. [Screenshot](https://user-images.githubusercontent.com/33323077/82896144-5a081a00-9f4d-11ea-8022-bcbd4e9b0506.png).

You simply have to structure your my_none to receive a Regex, a ruby Class and a simple pattern. 

- [ ] Next, my_map, when both a proc and a block are passed, is expected to execute just the proc and return a result. Consider an example like this. [Screenshot](https://user-images.githubusercontent.com/33323077/82896513-fe8a5c00-9f4d-11ea-851a-c90374de01a7.png)

- [ ] Also my_inject doesn't behave as expected when a symbol is passed to it. It rather throws an error; 'Please provide a symbol'. 

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: my_methods.rb [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      my_each do |key, value|
+        return_arr << yield(key, value)
+      end
+    else
+      my_each do |element|
+        return_arr << yield(element)
+      end
+    end
+    return_arr
+  end
```
Comment (shshamim63) [inline my_methods.rb]: `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- returns an Enumerator if no block is given.
___
Now the updated Part,
image
- executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
```ruby
my_proc = Proc.new { |num| num > 10 }
[11, 2, 3, 15].my_map(my_proc) {|num| num < 10 } # should return [true, false, false, true]
```

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :recycle: Changes Required :battery: 
___
Hi @vanheaven-ui :wave: 
You have almost covered all the changes. Readme is well organized. But my_map is not covering the instructions properly. Let's get to it,
___
### :spades: Mandatory Changes :spades: 
___
- Your my_map is not covering all the changes properly. Check out the comments below to complete it.
___
##### Please resolve the issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by lekegitrepo** [comment]
#### Hi @vanheaven-ui :1st_place_medal: :100:
#### Good job :+1:  this project is approved.

### _Reviewed by Temitayo Adeleke._

**Review by lekegitrepo — APPROVED** [review]
##### Status: Approved.

**Pull request merged**